finagle/buoyant/src/main/scala/com/twitter/finagle/buoyant/TlsClientConfig.scala

```diff
@@ -25,17 +26,34 @@ case class TlsClientConfig(
         SslClientEngineFactory.Param(Netty4ClientEngineFactory())
 
     case TlsClientConfig(_, Some(cn), certs, clientAuth) =>
-      val trustStreams = certs.getOrElse(Nil).map(new FileInputStream(_))
-      val certCollection = File.createTempFile("certCollection", null)
-      val f = new FileOutputStream(certCollection)
-      for (cert <- trustStreams) StreamIO.copy(cert, f)
-      f.flush()
-      f.close()
-      certCollection.deleteOnExit()
+      // map over the optional certs parameter - we want to pass
+      // `TrustCredentials.CertCollection` if we were given a list of certs,
+      // but `TrustCredentials.Unspecified` (rather than an empty cert
+      // collection file) if we were not.
+      val credentials = certs map { certs: Seq[String] =>
```

Comment:
If I recall correctly, it wouldn't compile without that annotation? I'll double-check though.